benchmark/Benchmark/Compare.hs

```diff
@@ -0,0 +1,45 @@
+module Benchmark.Compare
+       (main
+       ) where
+
+import Data.Text (Text)
+import Gauge.Main (bench, bgroup, defaultMain, whnf)
+import Text.Toml (parseTomlDoc)
+import Text.Toml.Types (Table)
+
+import Benchmark.Htoml (convertHtoml, decodeHtoml)
+import Benchmark.Tomland (codec)
+import Toml.Bi.Code (decode, runTomlCodec)
+import Toml.Parser (parse)
+import Toml.Type (TOML)
+
+import qualified Data.Text.IO as TIO
+
+
+-- | Benchmark.
+main :: IO ()
+main = do
+    (txt, htoml, tomland) <- preload
+    defaultMain
+        [ bgroup "tomland"
+            [ bench "parse"   $ whnf parse txt
+            , bench "convert" $ whnf (runTomlCodec codec) tomland
+            , bench "decode"  $ whnf (decode codec) txt
+            ]
+        , bgroup "htoml"
+            [ bench "parse"   $ whnf (parseTomlDoc "") txt
+            , bench "convert" $ whnf convertHtoml htoml
+            , bench "decode"  $ whnf decodeHtoml txt
+            ]
+        ]
+
+-- | Preload toml file, and parsed version for libraries.
+preload :: IO (Text, Table, TOML)
+preload = do
+    txt <- TIO.readFile "./benchmark/benchmark.toml"
+    case parseTomlDoc "" txt of
+        Left _      -> error "Htoml parsing failed"
+        Right htoml ->
+            case parse txt of
+                Left _        -> error "Tomland parsing failed"
+                Right tomland -> pure (txt, htoml, tomland)
```

Comment:
I feel like this pattern-matching of `Left` and `Right` can become messy with addition of more libraries... What if you do the following in `main` function?

1. Read content of the file into variable `txt`
2. Have variables: `tomlandVal`, `htomlVal` for intermediate representations of parsed types to values.

So the code will look like:

```haskell
txt <- readTomlExample
tomlandVal <- unsafeFromRight $ Tomland.parse txt
htomlVal   <- unsafeFromRight $ Htoml.parse txt
```

Probably something better instead of `Val` should be used as suffix...